docs/monthly-meeting/2022-10.md

```diff
@@ -0,0 +1,91 @@
+# Documentation Community Team Meeting (October 2022)
+
+- **Date:** 2022-10-03
+- **Time:** [19:30 UTC](https://arewemeetingyet.com/UTC/2022-10-03/19:30/Docs%20Meeting)
+- **This HackMD:** https://hackmd.io/@encukou/pydocswg1
+- [**Discourse thread**](https://discuss.python.org/t/19070) for October
+- **Calendar for future meetings:** (send your e-mail to Mariatta for a calendar invitation)
+- **How to participate:**
+  -  Go to [Google Meet](https://meet.google.com/dii-qrzf-wkw) and ask to be let in.
+  -  To edit notes, click the “pencil” or “split view” button on the [hackMD document]
+  -  (https://hackmd.io/@encukou/pydocswg1). You need to log in (e.g. with a GitHub account).
+
+By participating in this meeting, you are agreeing to abide by and uphold the [PSF Code of Conduct](https://www.python.org/psf/codeofconduct/).
+Please take a second to read through it!
+
+## Roll call
+
+- Petr Viktorin / `@encukou`
+- Adam T / `@AA-Turner`
+- Ezio Melotti / `@ezio-melotti`
+- Hugo van Kemenade / `@hugovk`
+- Mariatta / `mariatta`
+- Joannah / `nanjekyejoannah`
+- Alex Waygood / `@AlexWaygood`
+- CAM Gerlach / `@CAM-Gerlach`
+- Pradyun / `@pradyunsg`
+- Leam Hall
+
+## Quick updates - Introductions
+
+> 60 second updates on things you have been up to, questions you have, or developments you think people should know about. Please add yourself, and if you do not have an update to share, you can pass.
+
+- Videos for the Diataxis workshop were published!  https://discuss.python.org/t/recordings-available-for-python-docs-diataxis-workshop/19518
+
+
+## Discussion
+
+* (CAM) Should we enable Intersphinx support on the CPython 3.12 docs? It would primarily benefit porting/syncing docs from other places, like PEPs, importlib metadata/resources, and packaging, but could cause problems for downstreams (e.g. Linux distros) building the docs from source without network access. See [python/cpython#97785](https://github.com/python/cpython/pull/97785/).
+
+   * Allows single source of truth for definitions, etc.
+   * Not offline buildable
+   * Petr: would like Sphinx to provide a directory to reference local inv files to do offline builds
```

Comment:
If I understand what the goal is here, this is already possible. When you declare `intersphinx_mappings` in `conf.py`, you can pass a tuple to the second argument. Sphinx will walk through each element of the tuple and work with the first successfully-found `objects.inv`.

See "Multiple targets for the inventory" in the [`intersphinx` docs](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#confval-intersphinx_mapping).

Reply:
This is what we talked about, but Petr apparently wanted to propose something a little different.

Reply:
Interesting...I'm curious now what he's wanting.

Just about anything I can think of should be handle-able by some dynamic code in `conf.py`.

Reply:
The distro doesn't really “own” `conf.py`, and would have to patch it. That's ugly and fragile/tedious. (Also note that when building docs for offlline use, we want the Intersphinx links to point to other offline (locally installed) docs, rather than to the Web.)
So I would prefer an *external* way (e.g. file named by environment variable) to specify that a given URL should be replaced by a local file. But that sounds like a “selfish” request from a distro, so I'm wary of suggesting it before I have time to contribute an implementation.
However, it could pave the way to support for building & interlinking multiple docs projects locally, for offline use: if projects could specify the “publish URL” in `conf.py`, Sphinx could also *update* the URL→file mapping. Perhaps that would be useful more widely.

Reply:
Nod, I see what you mean -- the need is potentially general, so better for it to be an official Sphinx feature rather than a `conf.py` hack; and, better just overall for it not to be a `conf.py` hack.

Thanks for the details!

Reply:
As a short term approach, we could just read in some env variable in the `conf.py` and use that to set the file path—it isn't ideal, but it isn't _that_ hacky. Certainly long term I imagine it would be nice to have a better mechanism, but the timeline for distros adopting a Sphinx version with it might not be short or consistant with adopting the relevant Python version(s) with Intersphinx.

> if projects could specify the “publish URL” in conf.py, Sphinx could also update the URL→file mapping. 

Hmm, though projects would ultimately still need a way to find the other projects' `conf.py`s, and have some sort of mapping to them that the original project can specify. I imagine that's possible on the distro side, but I'm not sure if there's a generalized solution beyond that.

Reply:
FYI, see python/cpython#97781 for the initial motivating case

Reply:
Thinking about it some more, just being able to override `intersphinx_mapping` with some arbitrary locally set value would give you both things you want—you can set both the source `.inv` location and the target of the generated links to resolve to a local file location as well as a HTTP(S) URL. Right now, we could add a bit of code in `conf.py` that reads this value in from an env variable if present and `eval`s it, overriding the configured default.

A longer term Sphinx-level solution could perhaps be some way, e.g. a config file, to map URLs to source `.inv`s and target docs locations.

Reply:
I'll open an issue, PR and Discourse thread to summarize all this and continue discussion, as soon as I get a free moment during the sprints.

Reply:
IMO:
For the motivating example: if Python wants to adopt PyPA terminology in its own docs, we should copy the definitions over.
For distros (speaking with my Python hat on): don't worry too much about them. If they care, they can patch/workaround, and then iterate toward a better solution.